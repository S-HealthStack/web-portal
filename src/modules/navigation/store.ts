import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';

export enum Path {
  Root = '/',

  SignIn = '/signin',
  AccountActivation = '/account/activation',
  AccountCreate = '/account/create',
  AccountConfirm = '/account/confirm',
  AccountVerification = '/email-verification',
  ForgotPassword = '/account/forgot',
  ForgotPasswordConfirm = '/account/forgot/confirm',
  ResetPassword = '/account/forgot/reset',
  ResetPasswordComplete = '/account/forgot/complete',

  CreateStudy = '/create-study',

  Overview = '/overview',
  OverviewSubject = '/overview/participants/:subjectId',
  TrialManagement = '/trial-management',
  TrialManagementSubject = '/trial-management/participants/:subjectId',
  TrialManagementEditSurvey = '/trial-management/surveys/:surveyId/edit',
  TrialManagementSurveyResults = '/trial-management/surveys/:surveyId/results',
  UserAnalytics = '/user-analytics',
  DataCollection = '/data-collection',
  DataCollectionSubject = '/data-collection/participants/:subjectId',
  StudySettings = '/study-settings',
}

export const makeHistory = () =>
  process.env.NODE_ENV !== 'test'
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.OverviewSubject,
  Path.TrialManagement,
  Path.UserAnalytics,
  Path.DataCollection,
  Path.StudySettings,
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
