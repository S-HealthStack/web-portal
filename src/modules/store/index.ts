export type { RootState, AppDispatch, AppThunk } from './store';
export { useAppDispatch, useAppSelector } from './hooks';
export * from './tools';
