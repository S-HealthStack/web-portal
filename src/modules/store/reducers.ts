import { combineReducers } from '@reduxjs/toolkit';
import { connectRouter } from 'connected-react-router';
import type { Reducer } from 'redux';

import authReducer from 'src/modules/auth/auth.slice';
import studySettingsReducer from 'src/modules/study-settings/studySettings.slice';
import snackbarReducer from 'src/modules/snackbar/snackbar.slice';
import overviewReducers from 'src/modules/overview/overview.slice';
import studiesReducer from 'src/modules/studies/studies.slice';
import studyManagementReducer from 'src/modules/study-management/studyManagement.slice';
import dataCollectionReducer from 'src/modules/data-collection/dataCollection.slice';
import sidebarReducer from 'src/modules/main-layout/sidebar/sidebar.slice';
import fileUploadReducer from 'src/modules/file-upload/fileUpload.slice';
import { history } from 'src/modules/navigation/store';
import { DataSliceState } from 'src/modules/store/createDataSlice';
import forgotPasswordReducers from 'src/modules/auth/forgot-password/forgotPassword.slice';
import resetPasswordReducers from 'src/modules/auth/forgot-password/resetPassword.slice';
import objectStorageReducers from 'src/modules/object-storage/objectStorage.slice';

const createReducer = <T, V>(asyncReducers?: { [key: string]: Reducer<DataSliceState<T, V>> }) =>
  combineReducers({
    ...authReducer,
    router: connectRouter(history),
    snackbar: snackbarReducer,
    dataCollection: dataCollectionReducer,
    sidebar: sidebarReducer,
    ...studySettingsReducer,
    ...overviewReducers,
    ...studyManagementReducer,
    ...forgotPasswordReducers,
    ...resetPasswordReducers,
    ...fileUploadReducer,
    ...asyncReducers,
    ...objectStorageReducers,
    ...studiesReducer,
  });

export default createReducer;
