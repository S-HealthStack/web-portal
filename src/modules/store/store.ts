/* eslint-disable import/no-import-module-exports */
import {
  Action,
  ThunkAction,
  configureStore,
  getDefaultMiddleware,
  ThunkDispatch,
} from '@reduxjs/toolkit';
import { routerMiddleware } from 'connected-react-router';

import { history } from 'src/modules/navigation/store';
import rootReducer from 'src/modules/store/reducers';

export { rootReducer };

export const store = configureStore({
  reducer: rootReducer,
  middleware: [...getDefaultMiddleware(), routerMiddleware(history)],
  devTools: process.env.NODE_ENV !== 'production',
});

if (module.hot) {
  // Enable Webpack hot module replacement for reducers
  module.hot.accept('./reducers', async () => {
    // eslint-disable-next-line import/no-useless-path-segments
    const nextRootReducer = (await import('../store/reducers')).default;
    store.replaceReducer(nextRootReducer);
  });
}

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = ThunkDispatch<RootState, unknown, Action>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
