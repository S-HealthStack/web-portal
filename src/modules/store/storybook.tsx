import React from 'react';
import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { DecoratorFunction } from '@storybook/csf';
import { ReactFramework } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { rootReducer, RootState } from './store';

export const getStoreDecorator = (state: ConfigureStoreOptions<RootState>['preloadedState']) => {
  const store = configureStore({
    reducer: rootReducer,
    preloadedState: state,
  });

  return ((story) => (
    <Provider store={store}>
      <MemoryRouter>{story()}</MemoryRouter>
    </Provider>
  )) as DecoratorFunction<ReactFramework>;
};
