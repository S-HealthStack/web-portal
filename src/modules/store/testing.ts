import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { rootReducer, RootState } from './store';

export const createTestStore = (state: ConfigureStoreOptions<RootState>['preloadedState']) =>
  configureStore({
    reducer: rootReducer(),
    preloadedState: state,
  });
