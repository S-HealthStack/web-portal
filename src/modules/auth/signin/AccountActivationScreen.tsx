import React, { ChangeEvent, useEffect, useState } from 'react';
import useSearchParam from 'react-use/lib/useSearchParam';

import styled from 'styled-components';

import LockIcon from 'src/assets/icons/lock.svg';
import Button from 'src/common/components/Button';
import InputField from 'src/common/components/InputField';
import PasswordInputField from 'src/common/components/PasswordInputField';
import { SnackbarContainer } from 'src/modules/snackbar';
import { useAppDispatch } from 'src/modules/store';
import { px, typography } from 'src/styles';
import { activateAccount } from '../auth.slice';
import { useEnterPress } from './hooks';
import ScreenCenteredCard from '../common/ScreenCenteredCard';
import { RequirementItem, useCheckPassword } from '../common/PasswordRequirements';

const MainWrapper = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  @media (max-height: ${px(704)}) {
    overflow-y: scroll;
  }
  @media (max-width: ${px(732)}) {
    overflow-x: scroll;
  }
`;

const ContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
`;

const Content = styled.div`
  height: ${px(641)};
  width: ${px(448)};
  margin: auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

const Header = styled.div`
  ${typography.headingLargeSemibold};
  margin: 0 auto ${px(36)};
`;

const RequirementsWrapper = styled.div`
  display: inline-flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
  height: ${px(182)};
  width: ${px(447)};
  top: ${px(8)};
  margin-bottom: ${px(24)};
`;

const AccountActivationScreen: React.FC = () => {
  const [password, setPassword] = useState<string>('');
  const [disabled, setDisabled] = useState<boolean>(true);
  const [name, setName] = useState<string>('');
  const [isLoading, setLoading] = useState(false);

  const dispatch = useAppDispatch();

  const email = useSearchParam('email') || '';
  const resetToken = useSearchParam('reset-token') || '';

  // TODO: maybe later backend will provide user name
  const [isPasswordPassed, requirements] = useCheckPassword({
    password,
    name: email.split('@')[0] || '',
  });

  const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
    setPassword(event.target.value);
  };

  const handleNameChange = (event: ChangeEvent<HTMLInputElement>) => {
    setName(event.target.value);
  };

  const handleClick = async () => {
    if (disabled) {
      return;
    }
    try {
      setLoading(true);
      await dispatch(
        activateAccount({
          name,
          password,
          resetToken,
        })
      );
    } catch (err) {
      // handled by store
    } finally {
      setLoading(false);
    }
  };

  useEnterPress(handleClick);

  useEffect(() => {
    setDisabled(!password || !isPasswordPassed || !name);
  }, [password, isPasswordPassed, name]);

  return (
    <MainWrapper>
      <ScreenCenteredCard minWidth={733} width={55.556} ratio={0.96125}>
        <ContentWrapper>
          <Content>
            <Header>Account activation</Header>
            <InputField
              type="email"
              label="Email"
              value={email}
              disabled
              endExtra={{ component: <LockIcon />, extraWidth: 24 }}
              placeholder="Enter email"
            />
            <InputField
              type="text"
              label="Name"
              value={name}
              onChange={handleNameChange}
              placeholder="Enter name"
              maxLength={30}
            />
            <PasswordInputField
              error={password && !isPasswordPassed}
              label="Password"
              value={password}
              onChange={handlePasswordChange}
              placeholder="Enter password"
              maxLength={256}
            />
            <RequirementsWrapper>
              {requirements.map((requirement) => (
                <RequirementItem
                  key={requirement.rule}
                  rule={requirement.rule}
                  passed={!password ? undefined : requirement.passed}
                  disabled={!password}
                />
              ))}
            </RequirementsWrapper>
            <Button disabled={disabled} onClick={handleClick} fill="solid" $loading={isLoading}>
              Activate Account
            </Button>
          </Content>
          <SnackbarContainer />
        </ContentWrapper>
      </ScreenCenteredCard>
    </MainWrapper>
  );
};

export default AccountActivationScreen;
