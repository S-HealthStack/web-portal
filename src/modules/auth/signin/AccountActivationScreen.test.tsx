import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import theme from 'src/styles/theme';
import AccountActivationScreen from 'src/modules/auth/signin/AccountActivationScreen';
import { store } from 'src/modules/store/store';
import { history } from 'src/modules/navigation/store';

describe('AccountActivationScreen', () => {
  it('should render', () => {
    const { baseElement } = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <AccountActivationScreen />
          </ConnectedRouter>
        </Provider>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
