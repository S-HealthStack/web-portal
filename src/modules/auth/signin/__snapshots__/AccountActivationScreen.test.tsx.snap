// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountActivationScreen should render 1`] = `
.c28 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  pointer-events: none;
}

.c26 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: #4475E3;
}

.c26 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c26 > div svg {
  margin-right: 0.25rem;
}

.c26 > div {
  color: #FFFFFF;
}

.c26 > div svg {
  fill: #FFFFFF;
}

.c26:hover:enabled {
  background-color: #2F65E0;
}

.c26:disabled {
  background-color: #B3C6F1;
}

.c26:focus-visible:enabled {
  border: 0.125rem solid #FFFFFF;
  background-color: #4475E3;
  outline: 0.125rem solid #C7D6F7;
}

.c27 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c10 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c10 > svg path {
  fill: #A1A1A1;
}

.c14 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 0.5rem);
}

.c15 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c9 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c9:hover:enabled {
  border-color: #2F65E0;
}

.c9:active:enabled,
.c9:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c9:disabled {
  color: rgba(0,0,0,0.38);
  border-color: transparent;
  background-color: #f8f8f8;
}

.c9::-webkit-input-placeholder {
  color: #808080;
}

.c9::-moz-placeholder {
  color: #808080;
}

.c9:-ms-input-placeholder {
  color: #808080;
}

.c9::placeholder {
  color: #808080;
}

.c9:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c9:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c9:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c9:disabled::placeholder {
  color: #DBDBDB;
}

.c9:-webkit-autofill,
.c9:-webkit-autofill:hover,
.c9:-webkit-autofill:focus {
  border: 0.0625rem solid rgba(219,219,219,0.2);
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c9:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c13 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c13:hover:enabled {
  border-color: #2F65E0;
}

.c13:active:enabled,
.c13:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c13:disabled {
  color: rgba(0,0,0,0.38);
  border-color: transparent;
  background-color: #f8f8f8;
}

.c13::-webkit-input-placeholder {
  color: #808080;
}

.c13::-moz-placeholder {
  color: #808080;
}

.c13:-ms-input-placeholder {
  color: #808080;
}

.c13::placeholder {
  color: #808080;
}

.c13:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c13:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c13:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c13:disabled::placeholder {
  color: #DBDBDB;
}

.c13:-webkit-autofill,
.c13:-webkit-autofill:hover,
.c13:-webkit-autofill:focus {
  border: 0.0625rem solid #4475E3;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c13:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c7 {
  max-height: 3.5rem;
}

.c7:hover .c8:enabled {
  border-color: #2F65E0;
}

.c6 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: rgba(0,0,0,0.38);
  gap: 0.5rem;
  height: 1.125rem;
}

.c12 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  gap: 0.5rem;
  height: 1.125rem;
}

.c11 {
  font-family: Inter,sans-serif;
  font-size: 0.625rem;
  line-height: 0.9375rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  padding-left: 1rem;
  height: 1.0625rem;
}

.c16 {
  background: none;
  border: none;
  padding: 0;
  height: 1.5rem;
  width: 1.5rem;
}

.c16 svg {
  fill: #4475E3;
}

.c16:active,
.c16:enabled:hover {
  cursor: pointer;
}

.c16:active svg,
.c16:enabled:hover svg {
  fill: #4475E3;
}

.c16:disabled svg {
  fill: #DBDBDB;
}

.c17 {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c17:hover {
  cursor: pointer;
}

.c29 {
  position: relative;
  height: 0;
  width: 100%;
  pointer-events: none;
}

.c1 {
  color: #474747;
  min-width: 45.8125rem;
  width: 55.556vw;
  min-height: 44.037265625rem;
  height: 53.403205vw;
  max-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c21 {
  display: inline-grid;
  gap: 0.25rem;
  grid-template-columns: 2.5rem 1fr;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  cursor: unset !important;
}

.c24 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #838383;
  -webkit-transition: color 300ms cubic-bezier(0.45,0.05,0,1);
  transition: color 300ms cubic-bezier(0.45,0.05,0,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c23 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c20 {
  display: inline-block;
}

.c20 input {
  display: none;
}

.c20 .c22 > svg > :nth-child(1) {
  -webkit-transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  fill: #DBDBDB;
}

.c20 .c22 > svg > :nth-child(2) {
  -webkit-transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
}

.c19 {
  height: 2.5rem;
}

.c19 * {
  cursor: default;
}

.c25 {
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-style: normal;
  color: #838383;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  height: 40.0625rem;
  width: 28rem;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 2rem;
  line-height: 2.625rem;
  font-weight: 600;
  font-style: normal;
  margin: 0 auto 2.25rem;
}

.c18 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 11.375rem;
  width: 27.9375rem;
  top: 0.5rem;
  margin-bottom: 1.5rem;
}

@media (max-height:44rem) {
  .c0 {
    overflow-y: scroll;
  }
}

@media (max-width:45.75rem) {
  .c0 {
    overflow-x: scroll;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
        width="55.556"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <div
              class="c4"
            >
              Account activation
            </div>
            <div
              class="c5"
            >
              <div
                class="c6"
                data-testid="input-description"
              >
                Email
              </div>
              <div
                class="c7"
              >
                <input
                  class="c8 c9"
                  data-testid="input"
                  disabled=""
                  placeholder="Enter email"
                  type="email"
                  value=""
                />
                <div
                  class="c10"
                  disabled=""
                >
                  <svg />
                </div>
              </div>
              <div
                class="c11"
              >
                 
              </div>
            </div>
            <div
              class="c5"
            >
              <div
                class="c12"
                data-testid="input-description"
              >
                Name
              </div>
              <div
                class="c7"
              >
                <input
                  class="c8 c13"
                  data-testid="input"
                  maxlength="30"
                  placeholder="Enter name"
                  type="text"
                  value=""
                />
                <div
                  class="c14"
                />
              </div>
              <div
                class="c11"
              >
                 
              </div>
            </div>
            <div
              class="c5"
            >
              <div
                class="c12"
                data-testid="input-description"
              >
                Password
              </div>
              <div
                class="c7"
              >
                <input
                  class="c8 c13"
                  data-testid="input"
                  maxlength="256"
                  placeholder="Enter password"
                  type="password"
                  value=""
                />
                <div
                  class="c15"
                >
                  <button
                    aria-label="Show Password"
                    class="c16 c17"
                    color="primary"
                  >
                    <svg
                      data-testid="icon"
                    />
                  </button>
                </div>
              </div>
              <div
                class="c11"
              >
                 
              </div>
            </div>
            <div
              class="c18"
            >
              <div
                class="c19"
              >
                <div
                  class="c20 "
                  color="disabled"
                  disabled=""
                  kind="success"
                  readonly=""
                >
                  <label
                    class="c21"
                    for="1"
                    kind="success"
                  >
                    <div
                      class="c22 c23"
                    >
                      <svg />
                    </div>
                    <div
                      class="c24"
                      disabled=""
                    >
                      <span
                        class="c25"
                      >
                        Be at least 12 characters in length
                      </span>
                    </div>
                  </label>
                  <input
                    checked=""
                    data-testid="auth-signup-radio"
                    disabled=""
                    id="1"
                    readonly=""
                    type="checkbox"
                  />
                </div>
              </div>
              <div
                class="c19"
              >
                <div
                  class="c20 "
                  color="disabled"
                  disabled=""
                  kind="success"
                  readonly=""
                >
                  <label
                    class="c21"
                    for="2"
                    kind="success"
                  >
                    <div
                      class="c22 c23"
                    >
                      <svg />
                    </div>
                    <div
                      class="c24"
                      disabled=""
                    >
                      <span
                        class="c25"
                      >
                        Contains 1 upper case, 1 lower case, 1 number, and 1 special character
                      </span>
                    </div>
                  </label>
                  <input
                    checked=""
                    data-testid="auth-signup-radio"
                    disabled=""
                    id="2"
                    readonly=""
                    type="checkbox"
                  />
                </div>
              </div>
              <div
                class="c19"
              >
                <div
                  class="c20 "
                  color="disabled"
                  disabled=""
                  kind="success"
                  readonly=""
                >
                  <label
                    class="c21"
                    for="3"
                    kind="success"
                  >
                    <div
                      class="c22 c23"
                    >
                      <svg />
                    </div>
                    <div
                      class="c24"
                      disabled=""
                    >
                      <span
                        class="c25"
                      >
                        Does not contain more than 3 identical characters in a row
                      </span>
                    </div>
                  </label>
                  <input
                    checked=""
                    data-testid="auth-signup-radio"
                    disabled=""
                    id="3"
                    readonly=""
                    type="checkbox"
                  />
                </div>
              </div>
              <div
                class="c19"
              >
                <div
                  class="c20 "
                  color="disabled"
                  disabled=""
                  kind="success"
                  readonly=""
                >
                  <label
                    class="c21"
                    for="4"
                    kind="success"
                  >
                    <div
                      class="c22 c23"
                    >
                      <svg />
                    </div>
                    <div
                      class="c24"
                      disabled=""
                    >
                      <span
                        class="c25"
                      >
                        Does not contain username
                      </span>
                    </div>
                  </label>
                  <input
                    checked=""
                    data-testid="auth-signup-radio"
                    disabled=""
                    id="4"
                    readonly=""
                    type="checkbox"
                  />
                </div>
              </div>
            </div>
            <button
              class="c26"
              disabled=""
              fill="solid"
              style="overflow: hidden; position: relative;"
            >
              <div
                class="c27"
                data-testid="content"
              >
                Activate Account
              </div>
              <div
                class="c28"
                color="primaryPressed"
                opacity="1,1"
                scale="0"
                style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
              />
            </button>
          </div>
          <div
            class="c29"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
